students/MzKhan/lesson05/mailroom3.py

```diff
@@ -0,0 +1,177 @@
+donations = [ [6000.20, 2200, 1050,100],
+                [550,1000,250],
+                [55.66,270,1000],
+                [500, 500,200],
+                [12000.99, 5000],
+                [260.57, 930.90],
+                [6000, 2460.20, 900] ]
+
+# Create the dictionary using the dict comprehension.
+donors_data = dict(zip(donors_name,donations))
+
+def prompt_user():
+    """
+    prompt the user for the valid input.
+    input: string
+    return: int
+    """
+    print('Please choose one of the following options: \n')
+    print("1: Send a Thank You")
+    print("2: Create a report")
+    print("3: Send letters to everyone")
+    print("4: Quit")
+    return int(input("Option:  "))
+
+def thank_you():
+    """send a thank you note to a donor for his/her donation."""
+    name = name_prompt()
+    amount = donation_prompt(name)
+    add_donation(name, amount)
+    email(name, amount)
+
+def add_donation(name, amount):
+    """
+    parm: string
+    parm: float
+    """
+    donors_data.setdefault(name,[]).append(amount)
+
+def name_prompt():
+    """return a name string"""
+    donor_name = input("Please enter the donor's full name OR \n"
+                 +"type 'list' for the existing donors: ")
+
+    if not "".join(donor_name.split()).isalpha():
+        raise Exception("The name should be all Alphabetic characters: ")
+    if donor_name == 'list':
+        display_list()
+        donor_name = input("Please enter the donor's full name: ")
+        if not "".join(donor_name.split()).isalpha():
+            raise Exception("The name should be all Alphabetic characters: ")
+
+    return donor_name.title()
+
+def donation_prompt(name):
+    """
+    prompt the user for the donation amount for the given donor.
+    parm: string
+    return: float
+    """
+    donation_amount = input("Please enter the donation amount for {}: "
+                                                               .format(name))
+    return float(donation_amount)
+
+
+def display_list():
+    """print the donor's list to the user."""
+    print("\nOur generous donors: \n")
+    for donor in donors_data:
+        print(donor)
+
+def email(donor_name, amount):
+    """
+    generate the email for the given donor for his/her donation.
+    parm: string
+    parm: float
+    """
+    header = "\nDear {}".format(donor_name)
+    body = "\nThank you so much for your generous donation of $ {:.2f}."
+    print(header, body.format(amount))
+
+def print_report():
+    """Print the donors' report in the formatted tabular form."""
+    title = "{:24} | {:12} | {:10} | {:20}"
+    dashes=67*('-');print(dashes)
+    print(title.format('Donor Name','Total Given','Num Gifts','Average Gift'))
+    strf_format = "{:24} ${:12.2f}   {:^10d}  ${:12.2f}"
+    print(dashes)
+    report_list = calculate_total_gift()
+    for donor in sorted(report_list, key = igetter(1), reverse = True):
+        print(strf_format.format(*donor))
+    print(dashes)
+def calculate_total_gift():
```

Comment:
two lines space between module functions.

Reply:
I'm sorry I didn't understand what you mean by Two lines space between module functions?

Reply:
two lines space:
```
def function1():
    pass
# line 1 space
# line 2 space
def function2():
    pass
```
will look like:
```
def function1():
    pass


def function2():
    pass
```